site/server/app/com/fortysevendeg/exercises/persistence/domain/UserQueries.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * scala-exercises-server
+ * Copyright (C) 2015-2016 47 Degrees, LLC. <http://www.47deg.com>
+ */
+
+package com.fortysevendeg.exercises.persistence.domain
+
+import cats.data.Xor
+import shared.User
+import shapeless._
+import ops.record._
+
+object UserCreation {
+
+  // TODO: add proper case objects with errors when switching to Postgres
+  sealed trait CreationError
+
+  case object DuplicateName extends CreationError
+
+  case class Request(
+      login:      String,
+      name:       String,
+      githubId:   String,
+      pictureUrl: String,
```

Comment:
I have created a new empty account and I have two news: 

- The good one: GitHub provides an auto-generated avatar that it's not possible to be removed (only replaced).

- The bad one: When an unnamed user is logged the app throws an exception. This PR https://github.com/47deg/scala-exercises/pull/239 solves the problem. @juanpedromoreno @raulraja Could you take a look please?

The result:
![](http://rafp.es/1UfC4p2)